packages/editor-ui/src/composables/usePinnedData.ts

```diff
@@ -75,24 +75,36 @@ export function usePinnedData(
 		);
 	});
 
-	function canPinNode(checkDataEmpty = false) {
+	function canPinNode(checkDataEmpty = false, outputIndex?: number) {
 		const targetNode = unref(node);
-		if (targetNode === null) return false;
+		if (targetNode === null || PIN_DATA_NODE_TYPES_DENYLIST.includes(targetNode.type)) return false;
 
 		const nodeType = useNodeTypesStore().getNodeType(targetNode.type, targetNode.typeVersion);
 		const dataToPin = getInputDataWithPinned(targetNode);
 
 		if (!nodeType || (checkDataEmpty && dataToPin.length === 0)) return false;
 
 		const workflow = workflowsStore.getCurrentWorkflow();
-		const outputs = NodeHelpers.getNodeOutputs(workflow, targetNode, nodeType);
-		const mainOutputs = outputs.filter((output) =>
-			typeof output === 'string'
-				? output === NodeConnectionType.Main
-				: output.type === NodeConnectionType.Main,
+		const outputs = NodeHelpers.getNodeOutputs(workflow, targetNode, nodeType).map((output) =>
+			typeof output === 'string' ? { type: output } : output,
 		);
 
-		return mainOutputs.length === 1 && !PIN_DATA_NODE_TYPES_DENYLIST.includes(targetNode.type);
+		const mainOutputs = outputs.filter(
+			(output) => output.type === NodeConnectionType.Main && output.category !== 'error',
+		);
+
+		let indexAcceptable = true;
+
+		if (outputIndex !== undefined) {
+			const output = outputs[outputIndex];
+
+			if (outputs[outputIndex] === undefined)
+				throw new Error(`Out of bounds index ${outputIndex} provided to canPinNode`);
```

Comment:
This could brick the UI. So might be better to have a sensible default rather than break UI.

For example adding a node with no outputs.
<img width="1226" alt="Screenshot 2024-11-05 at 16 25 16" src="https://github.com/user-attachments/assets/0f57b953-5f45-44e6-9e81-e0fb844e75b0">

Reply:
Oh nice catch, I had assumed we always have an output.